### ISSUE TITLE
build fails with rust < 1.26

### ISSUE THREAD

**Issue by ThomasWaldmann** [benfred/py-spy#18]
on ubuntu 18.04 64bit:
```
$ sudo apt install cargo
...
The following NEW packages will be installed:
  cargo libgit2-26 libhttp-parser2.7.1 libstd-rust-1.25 libstd-rust-dev
  rust-gdb rustc
...

$ cargo install py-spy
    Updating registry `https://github.com/rust-lang/crates.io-index`
 Downloading py-spy v0.1.4                                                      
  Installing py-spy v0.1.4                                                      
 Downloading regex v1.0.5
 Downloading tempfile v3.0.3                                                    
 Downloading failure v0.1.2                                                     
 Downloading lazy_static v1.1.0                                                 
 Downloading clap v2.32.0                                                       
 Downloading proc-maps v0.1.4                                                   
 Downloading indicatif v0.9.0                                                   
 Downloading libc v0.2.43                                                       
 Downloading log v0.4.5                                                         
 Downloading goblin v0.0.15                                                     
 Downloading console v0.6.1                                                     
 Downloading env_logger v0.5.13                                                 
 Downloading tempdir v0.3.7                                                     
 Downloading benfred-read-process-memory v0.1.3                                 
 Downloading utf8-ranges v1.0.1                                                 
 Downloading memchr v2.0.2                                                      
 Downloading aho-corasick v0.6.8                                                
 Downloading thread_local v0.3.6                                                
 Downloading regex-syntax v0.6.2                                                
 Downloading version_check v0.1.4                                               
 Downloading ucd-util v0.1.1                                                    
 Downloading rand v0.5.5                                                        
 Downloading remove_dir_all v0.5.1                                              
 Downloading rand_core v0.2.1                                                   
 Downloading failure_derive v0.1.2                                              
 Downloading backtrace v0.3.9                                                   
 Downloading syn v0.14.9                                                        
 Downloading synstructure v0.9.0                                                
 Downloading quote v0.6.8                                                       
 Downloading proc-macro2 v0.4.18                                                
 Downloading unicode-xid v0.1.0                                                 
 Downloading rustc-demangle v0.1.9                                              
 Downloading cfg-if v0.1.5                                                      
 Downloading atty v0.2.11                                                       
 Downloading bitflags v1.0.4                                                    
 Downloading unicode-width v0.1.5                                               
 Downloading textwrap v0.10.0                                                   
 Downloading strsim v0.7.0                                                      
 Downloading vec_map v0.8.1                                                     
 Downloading lazy_static v0.2.11                                                
 Downloading parking_lot v0.6.4                                                 
 Downloading regex v0.2.11                                                      
 Downloading lock_api v0.1.3                                                    
 Downloading parking_lot_core v0.3.0                                            
 Downloading owning_ref v0.3.3                                                  
 Downloading scopeguard v0.3.3                                                  
 Downloading stable_deref_trait v1.1.1                                          
 Downloading smallvec v0.6.5                                                    
 Downloading unreachable v1.0.0                                                 
 Downloading void v1.0.2                                                        
 Downloading clicolors-control v0.2.0                                           
 Downloading regex-syntax v0.5.6                                                
 Downloading scroll v0.9.0                                                      
 Downloading plain v0.2.3                                                       
 Downloading scroll_derive v0.9.4                                               
 Downloading termcolor v1.0.3                                                   
 Downloading humantime v1.1.1                                                   
 Downloading quick-error v1.2.2                                                 
 Downloading rand v0.4.3                                                        
 Downloading log v0.3.9                                                         
 Downloading termios v0.2.2                                                     
 Downloading backtrace-sys v0.1.24                                              
 Downloading cc v1.0.23                                                         
 Downloading ansi_term v0.11.0                                                  
   Compiling regex v1.0.5                                                       
   Compiling stable_deref_trait v1.1.1
   Compiling rustc-demangle v0.1.9
   Compiling ansi_term v0.11.0
   Compiling scopeguard v0.3.3
   Compiling failure_derive v0.1.2
   Compiling ucd-util v0.1.1
   Compiling plain v0.2.3
   Compiling unicode-width v0.1.5
   Compiling bitflags v1.0.4
   Compiling remove_dir_all v0.5.1
   Compiling vec_map v0.8.1
   Compiling rand_core v0.2.1
   Compiling regex v0.2.11
   Compiling cc v1.0.23
   Compiling quick-error v1.2.2
   Compiling cfg-if v0.1.5
   Compiling strsim v0.7.0
   Compiling lazy_static v0.2.11
   Compiling version_check v0.1.4
   Compiling termcolor v1.0.3
   Compiling utf8-ranges v1.0.1
   Compiling libc v0.2.43
   Compiling unicode-xid v0.1.0
   Compiling void v1.0.2
   Compiling proc-macro2 v0.4.18
   Compiling owning_ref v0.3.3
   Compiling textwrap v0.10.0
   Compiling regex-syntax v0.6.2
   Compiling regex-syntax v0.5.6
   Compiling humantime v1.1.1
   Compiling log v0.4.5
   Compiling lazy_static v1.1.0
   Compiling atty v0.2.11
   Compiling rand v0.4.3
   Compiling clicolors-control v0.2.0
   Compiling memchr v2.0.2
   Compiling termios v0.2.2
   Compiling rand v0.5.5
   Compiling unreachable v1.0.0
   Compiling lock_api v0.1.3
   Compiling backtrace-sys v0.1.24
   Compiling log v0.3.9
   Compiling clap v2.32.0
   Compiling aho-corasick v0.6.8
   Compiling tempdir v0.3.7
   Compiling tempfile v3.0.3
   Compiling smallvec v0.6.5
   Compiling benfred-read-process-memory v0.1.3
   Compiling parking_lot_core v0.3.0
error[E0658]: use of unstable library feature 'thread_local_state': state querying was recently added (see issue #27716)
   --> /home/tw/.cargo/registry/src/github.com-1ecc6299db9ec823/parking_lot_core-0.3.0/src/parking_lot.rs:169:34
    |
169 |     if let Ok(tls) = THREAD_DATA.try_with(|x| x as *const ThreadData) {
    |                                  ^^^^^^^^

error[E0658]: use of unstable library feature 'thread_local_state': state querying was recently added (see issue #27716)
  --> /home/tw/.cargo/registry/src/github.com-1ecc6299db9ec823/parking_lot_core-0.3.0/src/word_lock.rs:55:38
   |
55 |         if let Ok(tls) = THREAD_DATA.try_with(|x| x as *const ThreadData) {
   |                                      ^^^^^^^^

error: aborting due to 2 previous errors

error: Could not compile `parking_lot_core`.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `py-spy v0.1.4`, intermediate artifacts can be found at `/tmp/cargo-install.SRMdEsZH4KOK`

Caused by:
  build failed
```

**Comment by kamyarrasta** [comment]
Exactly  the same issue on Ubuntu 16.04.5 LTS

**Comment by benfred** [comment]
@kamyarrasta I compile this on ubuntu 16.04 LTS all the time =)

What version of rust are you using? I'm guessing that that dependency is requiring a newer version of rust/

fwiw, I'm using rust 1.28

```
ben@ubuntu-xenial:~# lsb_release -a 
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.5 LTS
Release:	16.04
Codename:	xenial


ben@ubuntu-xenial:~# rustc --version
rustc 1.28.0 (9634041f0 2018-07-30)
```

**Comment by kamyarrasta** [comment]
Apparently I do have an old version of rust:
```
k@L:~# rustc --version
rustc 1.25.0

k@L:~# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.5 LTS
Release:        16.04
Codename:       xenial
```
I haven't check with other versions yet, but I will to see if it would work or not.

**Comment by ThomasWaldmann** [comment]
And in 18.04, they still have the same version...

**Comment by ThomasWaldmann** [comment]
https://github.com/Amanieu/parking_lot/issues/94

**Comment by ThomasWaldmann** [comment]
Installed rust 1.28 from the rust homepage now (they offer installer binaries and signatures there, so one can do a `/usr/local` install). Installing py-spy with that worked.

Couldn't find a rust ppa or backport or other newer package for ubuntu bionic. At least they have a fresher rust version for cosmic though (for the few installing non-lts versions).

**Comment by kamyarrasta** [comment]
I just have checked the build with rust 1.28 on ubuntu 16.04 and it works correctly now.

**Comment by benfred** [comment]
@ThomasWaldmann thanks for getting that fix into parking_lot_core - I tested out the new version of parking_lot_core (v0.3.1) and it compiles with rust v1.2.5. 

I've updated the Cargo.lock file in the last commit (https://github.com/benfred/py-spy/commit/23c0796ec6addf972ac7d8eb4ef393409cc64860), and with that change it looks like py-spy will compile with rust v1.2.5. 

This change will be in the version. Thanks everyone!
